python/rola/core.py

```diff
@@ -0,0 +1,52 @@
+from rola.models.signed_challenge import SignedChallenge
+from rola.utils.gateway import GatewayMetadataProvider
+from rola.utils.helpers import create_public_key_hash, create_signature_message
+from rola.utils.ret import derive_address
+
+
+class Rola:
+    def __init__(
+        self,
+        network_id: int,
+        dapp_address: str,
+        expected_origin: str,
+        application_name: str,
+        gateway_metadata_provider: GatewayMetadataProvider,
+    ):
+        self.network_id = network_id
+        self.dapp_address = dapp_address
+        self.expected_origin = expected_origin
+        self.application_name = application_name
+        self.gateway_metadata_provider = gateway_metadata_provider
+
+    def verify_signed_challenge(self, signed_challenge: SignedChallenge) -> bool:
+        # create public key hex hash
+        public_key_hash_hex = create_public_key_hash(signed_challenge.proof.public_key)
```

Comment:
Incompatiable type: `signed_challenge.proof.public_key` is `bytes` and `create_public_key_hash` takes `str` arguments. Perhaps we should do a mypy check in CI?

Reply:
fixed. added mypy to ci